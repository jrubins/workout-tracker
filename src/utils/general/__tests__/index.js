import { stringToObject } from '../'

describe('general utils', () => {
  describe('stringToObject', () => {
    const defaultOpts = {
      propDelimiter: ':',
      value: 'foo,bar:blah,ket',
      valueDelimiter: ',',
    }

    test('returns an empty object if no value is provided', () => {
      expect(
        stringToObject({
          value: null,
        })
      ).toEqual({})
    })

    test('parses the string into an object with delimiters', () => {
      expect(stringToObject(defaultOpts)).toMatchSnapshot()
    })

    test('allows a value processor to be specified', () => {
      expect(
        stringToObject({
          ...defaultOpts,
          valueProcessorFn: value => value + 1,
        })
      ).toMatchSnapshot()
    })
  })
})
