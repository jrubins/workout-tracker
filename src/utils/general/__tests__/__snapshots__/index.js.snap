// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`general utils stringToObject allows a value processor to be specified 1`] = `
Object {
  "blah": "ket1",
  "foo": "bar1",
}
`;

exports[`general utils stringToObject parses the string into an object with delimiters 1`] = `
Object {
  "blah": "ket",
  "foo": "bar",
}
`;
